From 016433d4ddaa5acfdc6b7d1b36a27123c31fa12c Mon Sep 17 00:00:00 2001
From: Steven Veigas <sveiga@codeaurora.org>
Date: Thu, 20 Nov 2014 14:26:20 +0000
Subject: [PATCH 2/2] lk: disabled nand flash support

for automotive platforms, disabled nand support for boot time
improvements

Change-Id: I069945b8649c1de42782775c0aced9ea8134316b
---
 app/aboot/aboot.c            | 27 ++++++++++++++++++++++++---
 app/aboot/recovery.c         |  7 ++++++-
 platform/msm_shared/rules.mk |  1 -
 3 files changed, 30 insertions(+), 5 deletions(-)

diff --git a/app/aboot/aboot.c b/app/aboot/aboot.c
index dab78d4..669a2ba 100644
--- a/app/aboot/aboot.c
+++ b/app/aboot/aboot.c
@@ -37,8 +37,9 @@
 #include <string.h>
 #include <kernel/thread.h>
 #include <arch/ops.h>
-
+#ifdef NAND_FLASH_SUPPORT
 #include <dev/flash.h>
+#endif
 #include <lib/ptable.h>
 #include <dev/keys.h>
 #include <dev/fbcon.h>
@@ -354,6 +355,7 @@ unsigned *atag_ramdisk(unsigned *ptr, void *ramdisk,
 	return ptr;
 }
 
+#ifdef NAND_FLASH_SUPPORT
 unsigned *atag_ptable(unsigned **ptr_addr)
 {
 	int i;
@@ -369,6 +371,7 @@ unsigned *atag_ptable(unsigned **ptr_addr)
 
 	return (*ptr_addr);
 }
+#endif
 
 unsigned *atag_cmdline(unsigned *ptr, const char *cmdline)
 {
@@ -405,10 +408,12 @@ void generate_atags(unsigned *ptr, const char *cmdline,
 	ptr = atag_ramdisk(ptr, ramdisk, ramdisk_size);
 	ptr = target_atag_mem(ptr);
 
+#ifdef NAND_FLASH_SUPPORT
 	/* Skip NAND partition ATAGS for eMMC boot */
 	if (!target_is_emmc_boot()){
 		ptr = atag_ptable(&ptr);
 	}
+#endif
 
 	ptr = atag_cmdline(ptr, cmdline);
 	ptr = atag_end(ptr);
@@ -815,6 +820,7 @@ unified_boot:
 
 int boot_linux_from_flash(void)
 {
+#ifdef NAND_FLASH_SUPPORT
 	struct boot_img_hdr *hdr = (void*) buf;
 	struct ptentry *ptn;
 	struct ptable *ptable;
@@ -1050,7 +1056,7 @@ continue_boot:
 	boot_linux((void *)hdr->kernel_addr, (void *)hdr->tags_addr,
 		   (const char *)cmdline, board_machtype(),
 		   (void *)hdr->ramdisk_addr, hdr->ramdisk_size);
-
+#endif
 	return 0;
 }
 
@@ -1113,6 +1119,7 @@ void read_device_info_mmc(device_info *dev)
 	memcpy(dev, info, sizeof(device_info));
 }
 
+#ifdef NAND_FLASH_SUPPORT
 void write_device_info_flash(device_info *dev)
 {
 	struct device_info *info = (void *) info_buf;
@@ -1177,6 +1184,7 @@ void read_device_info_flash(device_info *dev)
 	}
 	memcpy(dev, info, sizeof(device_info));
 }
+#endif
 
 void write_device_info(device_info *dev)
 {
@@ -1184,10 +1192,12 @@ void write_device_info(device_info *dev)
 	{
 		write_device_info_mmc(dev);
 	}
+#ifdef NAND_FLASH_SUPPORT
 	else
 	{
 		write_device_info_flash(dev);
 	}
+#endif
 }
 
 void read_device_info(device_info *dev)
@@ -1196,10 +1206,12 @@ void read_device_info(device_info *dev)
 	{
 		read_device_info_mmc(dev);
 	}
+#ifdef NAND_FLASH_SUPPORT
 	else
 	{
 		read_device_info_flash(dev);
 	}
+#endif
 }
 
 void reset_device_info()
@@ -1377,6 +1389,7 @@ void cmd_boot(const char *arg, void *data, unsigned sz)
 		   (void*) hdr->ramdisk_addr, hdr->ramdisk_size);
 }
 
+#ifdef NAND_FLASH_SUPPORT
 void cmd_erase(const char *arg, void *data, unsigned sz)
 {
 	struct ptentry *ptn;
@@ -1400,6 +1413,7 @@ void cmd_erase(const char *arg, void *data, unsigned sz)
 	}
 	fastboot_okay("");
 }
+#endif
 
 
 void cmd_erase_mmc(const char *arg, void *data, unsigned sz)
@@ -1626,6 +1640,7 @@ void cmd_flash_mmc(const char *arg, void *data, unsigned sz)
 	return;
 }
 
+#ifdef NAND_FLASH_SUPPORT
 void cmd_flash(const char *arg, void *data, unsigned sz)
 {
 	struct ptentry *ptn;
@@ -1671,6 +1686,7 @@ void cmd_flash(const char *arg, void *data, unsigned sz)
 	dprintf(INFO, "partition '%s' updated\n", ptn->name);
 	fastboot_okay("");
 }
+#endif
 
 void cmd_continue(const char *arg, void *data, unsigned sz)
 {
@@ -1726,6 +1742,7 @@ void splash_screen ()
 	struct ptable *ptable;
 	struct fbcon_config *fb_display = NULL;
 
+#ifdef NAND_FLASH_SUPPORT
 	if (!target_is_emmc_boot())
 	{
 		ptable = flash_get_ptable();
@@ -1748,6 +1765,7 @@ void splash_screen ()
 			}
 		}
 	}
+#endif
 }
 
 void aboot_init(const struct app_descriptor *app)
@@ -1762,11 +1780,13 @@ void aboot_init(const struct app_descriptor *app)
 		page_size = 2048;
 		page_mask = page_size - 1;
 	}
+#if NAND_FLASH_SUPPORT
 	else
 	{
 		page_size = flash_page_size();
 		page_mask = page_size - 1;
 	}
+#endif
 
 	ASSERT((MEMBASE + MEMSIZE) > MEMBASE);
 
@@ -1849,12 +1869,13 @@ fastboot:
 		fastboot_register("flash:", cmd_flash_mmc);
 		fastboot_register("erase:", cmd_erase_mmc);
 	}
+#ifdef NAND_FLASH_SUPPORT
 	else
 	{
 		fastboot_register("flash:", cmd_flash);
 		fastboot_register("erase:", cmd_erase);
 	}
-
+#endif
 	fastboot_register("continue", cmd_continue);
 	fastboot_register("reboot", cmd_reboot);
 	fastboot_register("reboot-bootloader", cmd_reboot_bootloader);
diff --git a/app/aboot/recovery.c b/app/aboot/recovery.c
index fc14a7f..c4eef1c 100644
--- a/app/aboot/recovery.c
+++ b/app/aboot/recovery.c
@@ -58,6 +58,7 @@ extern void set_device_root();
 
 int get_recovery_message(struct recovery_message *out)
 {
+#ifdef NAND_FLASH_SUPPORT
 	struct ptentry *ptn;
 	struct ptable *ptable;
 	unsigned offset = 0;
@@ -82,11 +83,13 @@ int get_recovery_message(struct recovery_message *out)
 		return -1;
 	}
 	memcpy(out, buf, sizeof(*out));
+#endif
 	return 0;
 }
 
 int set_recovery_message(const struct recovery_message *in)
 {
+#ifdef NAND_FLASH_SUPPORT
 	struct ptentry *ptn;
 	struct ptable *ptable;
 	unsigned offset = 0;
@@ -120,6 +123,7 @@ int set_recovery_message(const struct recovery_message *in)
 		dprintf(CRITICAL, "ERROR: flash write fail!\n");
 		return -1;
 	}
+#endif
 	return 0;
 }
 
@@ -200,6 +204,7 @@ int update_firmware_image (struct update_header *header, char *name)
 
 static int set_ssd_radio_update (char *name)
 {
+#ifdef NAND_FLASH_SUPPORT
 	struct ptentry *ptn;
 	struct ptable *ptable;
 	unsigned int ssd_cookie[2] = {0x53534443, 0x4F4F4B49};
@@ -225,7 +230,7 @@ static int set_ssd_radio_update (char *name)
 		dprintf(CRITICAL, "ERROR: flash write fail!\n");
 		return -1;
 	}
-
+#endif
 	dprintf(INFO, "FOTA partition written successfully!");
 	return 0;
 }
diff --git a/platform/msm_shared/rules.mk b/platform/msm_shared/rules.mk
index dc95091..c294fa5 100644
--- a/platform/msm_shared/rules.mk
+++ b/platform/msm_shared/rules.mk
@@ -13,7 +13,6 @@ OBJS += \
 	$(LOCAL_DIR)/hsusb.o \
 	$(LOCAL_DIR)/jtag_hook.o \
 	$(LOCAL_DIR)/jtag.o \
-	$(LOCAL_DIR)/nand.o \
 	$(LOCAL_DIR)/mmc.o \
 	$(LOCAL_DIR)/partition_parser.o
 
-- 
1.8.2.1

